File: benchmarks/pandas/bench_dataframe_cumops_axis1.py
"""
Benchmark: DataFrame.cumsum(axis=1) / cumprod(axis=1) (row-wise) on 10k x 8 DataFrame.
Outputs JSON: {"function": "dataframe_cumops_axis1", "mean_ms": ..., "iterations": ..., "total_ms": ...}
"""
import json
import time
import numpy as np
import pandas as pd

ROWS = 10_000
COLS = 8
WARMUP = 3
ITERATIONS = 20

data = {f"col{c}": ((np.arange(ROWS) + c) % 10) * 0.1 + 1 for c in range(COLS)}
df = pd.DataFrame(data)

for _ in range(WARMUP):
    df.cumsum(axis=1)
    df.cumprod(axis=1)

start = time.perf_counter()
for _ in range(ITERATIONS):
    df.cumsum(axis=1)
    df.cumprod(axis=1)
total = (time.perf_counter() - start) * 1000

print(json.dumps({"function": "dataframe_cumops_axis1", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))

File: benchmarks/tsb/bench_dataframe_cumops_axis1.ts
/**
 * Benchmark: dataFrameCumsum / dataFrameCumprod with axis=1 (row-wise) on 10k x 8 DataFrame.
 * Outputs JSON: {"function": "dataframe_cumops_axis1", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { DataFrame, dataFrameCumsum, dataFrameCumprod } from "../../src/index.ts";

const ROWS = 10_000;
const COLS = 8;
const WARMUP = 3;
const ITERATIONS = 20;

const data: Record<string, number[]> = {};
for (let c = 0; c < COLS; c++) {
  data[`col${c}`] = Array.from({ length: ROWS }, (_, i) => ((i + c) % 10) * 0.1 + 1);
}
const df = new DataFrame(data);

for (let i = 0; i < WARMUP; i++) {
  dataFrameCumsum(df, { axis: 1 });
  dataFrameCumprod(df, { axis: 1 });
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  dataFrameCumsum(df, { axis: 1 });
  dataFrameCumprod(df, { axis: 1 });
}
const total = performance.now() - start;

console.log(JSON.stringify({ function: "dataframe_cumops_axis1", mean_ms: total / ITERATIONS, iterations: ITERATIONS, total_ms: total }));
